{githubcsuser} iP

githubcsuser opened the pull request:
# Hello, I am **Taskmax** 🤖❤️
 

> "I am your personal task scheduling companion. 
My purpose is to **keep you organised and stress-free.** 
On a scale of 1 to 10, how would you rate your productivity?"  
> – **Taskmax**  

Taskmax is inspired by **Baymax** 🩺 from *Big Hero 6*! But instead of healthcare, I take care of your **tasks and deadlines** 🗓️.  

## 🎯 My Features:
1. 🤖 **List** all your tasks (**Persistent memory**)  
2. 📝 **Add** a task you plan to do  
3. ⏳ **Set deadlines** for your tasks  
4. 📆 **Schedule events**  
5. ❌ **Delete** tasks  
6. ✅ **Mark** tasks as complete  
7. 🔄 **Unmark** tasks that are incomplete  
8. :mag: **Find** a task containing a keyword

## 📌 Install me:
- [Download Taskmax here](https://github.com/githubcsuser/ip/releases).  
- **Double-click** to launch the application.  
- Start scheduling with your personal productivity bot!  

## 💡 Commands:
Use these commands to interact with **Taskmax**:

```bash
list                  # Displays all tasks  
todo [task]           # Adds a new task  
deadline [task] /by [date ime]  # Adds a task with a deadline  
event [task] /from [date time] /to [date time]  # Adds an event  
mark [task number]    # Marks a task as completed  
unmark [task number] #Unmarks an uncompleted task
delete [task number]  # Deletes a task  
find [keyword] #Finds a task containing the keyword
```

## ✅ What I can offer:
- [x] Ensure user satisfaction  
- [x] Store tasks persistently  
- [ ] Add recurring tasks (coming soon)  
- [ ] Smart notifications (coming soon)  

## ✨ Why me:
- **I am not fast** 🏃‍♂️, **but I am helpful.**  
- *On a scale of 1 to 10, how would you rate your task management?*  
- ~~Task anxiety is no longer a problem!~~  
- Inline code example: Use `todo Save the city` to add a task.  

> *"I will always be with you."* – **Taskmax** 🤖

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
level-0

Commit:
level-1

Commit:
level-1

Commit:
level-1.2

Commit:
level-2

Commit:
level-3

Commit:
level-4

Commit:
A-TextUiTesting

Commit:
level-5

Commit:
level-6

Commit:
level-6.2

Commit:
A-Enums

Commit:
typo

Commit:
Add Save

Commit:
Merge branch 'branch-Level-7'

Commit:
Update Deadline.java

Commit:
Update Event.java

Commit:
Add dates and times

Commit:
Merge branch 'branch-Level-8'

Commit:
Complete A-MoreOOP

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
Complete A-Packages

Commit:
Merge branch 'branch-A-Pacakages'

Commit:
Merge branch 'add-gradle-support'

Commit:
Add Gradle support (A-Gradle)

Commit:
Add JUnit tests for Task, Parser, Storage and TaskList

Commit:
Merge branch 'branch-A-JUnit'

Commit:
Update Gradle to generate JAR file

Commit:
Merge branch 'branch-A-Jar'

Commit:
Add JavaDoc comments to all classes

Commit:
Refactor code to follow coding standard

Commit:
Implement find command

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Resolve merge conflict for coding standard

Commit:
Add Level-9

Hackari commented on src/main/java/taskmax/main/Taskmax.java:
Line 11: Is there any particular reason why this line is separate from the other imports? (Lines 5 to 9) It is inconsistent in different files. For example, MarkCommand.java has combined them all:

```java
package taskmax.command;

import taskmax.exception.TaskmaxException;
import taskmax.storage.Storage;
import taskmax.storage.TaskList;
import taskmax.ui.Ui;
```

While Parser.java has separated them by subpackages:

```java
package taskmax.parser;

import taskmax.command.AddCommand;
import taskmax.command.Command;
import taskmax.command.DeleteCommand;
import taskmax.command.ExitCommand;
import taskmax.command.ListCommand;
import taskmax.command.MarkCommand;
import taskmax.command.UnmarkCommand;
import taskmax.command.FindCommand;

import taskmax.exception.TaskmaxException;

import taskmax.task.Deadline;
import taskmax.task.Event;
import taskmax.task.ToDo;

import taskmax.ui.Ui;
```

I think I like how it is implemented in Parser.java more.

githubcsuser replied:
I must have overlooked the imports in command, I intended to separate it like in Parser.java thank you

Hackari commented on src/main/java/taskmax/parser/Parser.java:
Based on the Java Coding guide, switch statements should have the following indentation:

```java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

Should be an easy fix for lines 37-76 though! Just Ctrl-Tab the entire contents of the Switch block.

Hackari commented on src/main/java/taskmax/storage/Storage.java:
Looks like there are some merge conflicts in the Javadocs 😲. It would be good to fix them!

Hackari commented on src/main/java/taskmax/task/Deadline.java:
From the Java Coding standard:
```
You MUST write header comments for all classes, public methods. But they can be omitted for the following cases:
i. Getters/setters
ii. When overriding methods (provided the parent method's Javadoc applies exactly as is to the overridden method)
```

For the toString method, I think it may be okay if there is no JavaDoc. This will apply to the other types of tasks too.

Hackari commented on src/main/java/taskmax/ui/Mascot.java:
That is one BIG mascot.

githubcsuser replied:
BIGGEST mascot for the BEST companionship 😄

Hackari commented on src/main/java/taskmax/ui/Ui.java:
This is not written explicitly in the coding standard, but I think it would improve readability:

Original at Lines 25 and 26:
```java
        System.out.println(LINE + intro + LINE + new Mascot() + LINE
                + "\nEnter \"hello!\" to begin\n" + LINE);
```
Suggestion:
```java
        System.out.println(LINE + intro + LINE + new Mascot() + LINE + "\n"
                + "Enter \"hello!\" to begin\n" 
                + LINE);
```

By ending each code line with a \n, it makes it very clear what the output might look like. This could apply to intro and new Mascot() too!

githubcsuser replied:
I agree with this comment

Hackari commented on src/test/java/taskmax/storage/StorageTest.java:
I believe the imports from Standard Java should be placed above other imports, so Lines 7 to 9 will now be above Lines 3 to 5. Don't fix this now though, I found it much easier to fix all these problems in one go after I had run gradle checkstyle to automatically flag out all coding standard errors. 😄

Hackari commented on src/test/java/taskmax/storage/TaskListTest.java:
Wildcard import! 😱 In this case, you'd only need Assertions.assertTrue, as used in line 18

githubcsuser replied:
oh no! I must have missed it!

Hackari commented on text-ui-test/EXPECTED.TXT:
That is one big companion. It even looks like a hero! Maybe you should add 6 of them.

githubcsuser replied:
😉

Hackari reviewed (commented):
Cool work! There are only a few nitpicky problems with the coding standard, and these are issues that can be fixed very quickly once the gradle checkstyle report generation is added!

Commit:
Add checkstyle

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Update Level-10

Commit:
Merge branch 'branch-Level-10'

Commit:
Add Varargs

Commit:
Merge branch 'A-Varargs'

Commit:
Parser, TaskList: Add assertions to enforce constraints

Parser and TaskList do not enforce certain constraints on user input
and task operations. This can result in unexpected behavior if invalid
indices or null inputs are processed.

Let's add assertions to:
* Ensure task indices are within valid bounds before performing operations.
* Validate user input is not null or empty before parsing.

This helps catch logical errors early and improves code robustness.

githubcsuser committed:
Merge pull request #1 from githubcsuser/branch-A-Assertions

Parser, TaskList: Add assertions to enforce constraints

Commit:
Refactor task management: Improve code structure and error handling

The task management code was refactored to improve maintainability and
readability. Key changes include:

* Split large classes into smaller, more manageable units
* Refined method names for clarity and consistency
* Consolidated redundant functionality into utility methods
* Enhanced exception handling with custom error messages
* Added input validation to ensure data integrity

These improvements will make the codebase easier to extend and debug,
with better error reporting and a clearer structure for future development.

githubcsuser committed:
Merge pull request #2 from githubcsuser/branch-A-CodeQuality

Refactor task management: Improve code structure and error handling

Commit:
TaskList: Refactor markTask to use Stream for task marking

The markTask method originally used direct index access to mark tasks as done.
This method has now been refactored to use Stream for finding and marking the task
at the given index.

Let's:
* Replace the direct index access with Stream's skip() and findFirst() methods.
* Ensure that the task is marked as done or not done based on the boolean flag.

This change introduces Streams, but it's less efficient than direct list access.
However, it aligns with the requirement to use Streams where applicable without
compromising the overall structure of the class.

githubcsuser committed:
Merge pull request #3 from githubcsuser/branch-A-Streams

TaskList: Refactor markTask to use Stream for task marking

Commit:
CI: Add GitHub Actions workflow for Gradle

This commit adds a GitHub Actions workflow to automate the CI process. The workflow runs on Ubuntu, macOS, and Windows, setting up JDK 17 and validating the Gradle wrapper. It also performs a Gradle build and runs I/O redirection tests if applicable.

This ensures automated testing on every push and pull request.

Commit:
Update CI workflow: Remove unnecessary I/O redirection tests

githubcsuser committed:
Merge pull request #4 from githubcsuser/branch-A-CI

CI: Add GitHub Actions workflow for Gradle

Commit:
Add sort by priority feature

githubcsuser committed:
Merge pull request #5 from githubcsuser/branch-BCD-Extension

Add sort by priority feature

Commit:
Update Taskmax personality

githubcsuser committed:
Merge pull request #7 from githubcsuser/branch-A-Personality

Update Taskmax personality

Commit:
Update error handling

githubcsuser committed:
Merge pull request #8 from githubcsuser/branch-A-MoreErrorHandling

Update error handling

Commit:
Update javadoc

Commit:
Update testing

Commit:
Merge branch 'branch-A-MoreTesting'

Commit:
Update user guide

Commit:
Update README.md

Commit:
update build.gradle

Commit:
Update main window dimensions

Commit:
Update javadocs

githubcsuser committed:
Update README.md

Commit:
Update README.md

githubcsuser committed:
Update README.md
